cartographer/mapping/pose_graph_data.h

```diff
@@ -74,6 +74,7 @@ struct PoseGraphData {
   // Global landmark poses with all observations.
   std::map<std::string /* landmark ID */, PoseGraphInterface::LandmarkNode>
       landmark_nodes;
+  std::set<std::string /* landmark ID */> frozen_landmarks;
```

Comment:
I like this change and it makes perfect sense to have control over what pose graph nodes are constant or not. Thank you for making this. 

Just one suggestion: instead of having a separate `set<string> frozen_landmarks` in `PoseGraphData`, please, add a field `bool frozen` to `PoseGraphInterface::LandmarkNode`. It should simplify the code and keep things together.

Reply:
Yes, I was thinking about that solution too. I can make the change eventually; however, it might take me a couple of weeks until I have time